dialogue-core/src/main/java/com/palantir/dialogue/core/DeprecationWarningChannel.java

```diff
@@ -73,19 +75,19 @@ public ListenableFuture<Response> execute(Request request) {
     }
 
     private FutureCallback<Response> createCallback(String channelName, Endpoint endpoint) {
-        Meter meter = metrics.deprecations(endpoint.serviceName());
-
+        // lazily create meter metric name only if deprecated endpoint is accessed
+        Supplier<Meter> meterSupplier = Suppliers.memoize(() -> metrics.deprecations(endpoint.serviceName()));
```

Comment:
I imagine this came up with something that used sticky sessions? iirc each sticky session client re-creates endpointchannel instances, which is why we memoize some other metrics.

Reply:
yep, I saw this on some of the major clients that use sticky sessions